### ISSUE TITLE
Does detect.py can provide the bounding box coordinate in the original image scale?

### ISSUE THREAD

**Issue by user54353** [ultralytics/yolov5#1893]
## ❔Question
How can I get the coordinates of the detected bounding box in terms of the original image scale?
Namely how to convert Yolo format bounding box coordinates into an original image scale? Can detect.py provide it?

**Comment by glenn-jocher** [comment]
@user54353 detections may be saved in normalized coordinates with `python detect.py --save-txt`. You can denormalize by multiplying x and width by image width and y and height by image height.

Alternatively you can use the PyTorch Hub implementation, and then request results directly in image coordinates:
```python
import cv2
import torch
from PIL import Image

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s', pretrained=True)

# Images
for f in ['zidane.jpg', 'bus.jpg']:  # download 2 images
    print(f'Downloading {f}...')
    torch.hub.download_url_to_file('https://github.com/ultralytics/yolov5/releases/download/v1.0/' + f, f)
img1 = Image.open('zidane.jpg')  # PIL image
img2 = cv2.imread('bus.jpg')[:, :, ::-1]  # OpenCV image (BGR to RGB)
imgs = [img1, img2]  # batched list of images

# Inference
results = model(imgs, size=640)  # includes NMS

# Results
results.print()  # print results to screen
results.show()  # display results
results.save()  # save as results1.jpg, results2.jpg... etc.

# Data
print('\n', results.xyxy[0])  # print img1 predictions in image coordinates
#          x1 (pixels)  y1 (pixels)  x2 (pixels)  y2 (pixels)   confidence        class
# tensor([[7.47613e+02, 4.01168e+01, 1.14978e+03, 7.12016e+02, 8.71210e-01, 0.00000e+00],
#         [1.17464e+02, 1.96875e+02, 1.00145e+03, 7.11802e+02, 8.08795e-01, 0.00000e+00],
#         [4.23969e+02, 4.30401e+02, 5.16833e+02, 7.20000e+02, 7.77376e-01, 2.70000e+01],
#         [9.81310e+02, 3.10712e+02, 1.03111e+03, 4.19273e+02, 2.86850e-01, 2.70000e+01]])

```

See https://pytorch.org/hub/ultralytics_yolov5/

**Comment by divinit7** [comment]
Hello, I'm having issues converting saved text files into bounding boxes, they do not map to correct position on the image . The image saved in the runs directory shows much higher confidence on the bounding box while pytorch hub loading shows really low confidence on the same bounding box. 
`python3 detect.py --save-txt --weights ../best.pt --imgsz 1280 --source ../key-extraction-images --conf-thres 0.7`

**Comment by glenn-jocher** [comment]
@divinit7  detect.py operates correctly and saves text file labels in **YOLO format**, with one `*.txt` file per image (if no objects in image, no `*.txt` file is required). The `*.txt` file specifications are:

- One row per object
- Each row is `class x_center y_center width height` format.
- Box coordinates must be in **normalized xywh** format (from 0 - 1). If your boxes are in pixels, divide `x_center` and `width` by image width, and `y_center` and `height` by image height.
- Class numbers are zero-indexed (start from 0).

<img width="800" alt="Image Labels" src="https://user-images.githubusercontent.com/26833433/91506361-c7965000-e886-11ea-8291-c72b98c25eec.jpg">

The label file corresponding to the above image contains 2 persons (class `0`) and a tie (class `27`):

<p align="center"><img width="428" src="https://user-images.githubusercontent.com/26833433/112467037-d2568c00-8d66-11eb-8796-55402ac0d62f.png"></p>

Good luck 🍀 and let us know if you have any other questions!
